app/app.go

```diff
@@ -543,38 +531,47 @@ func NewMigalooApp(
 		app.Ics20WasmHooks,
 	)
 
-	// IBC Fee Module keeper
-	app.IBCFeeKeeper = ibcfeekeeper.NewKeeper(
-		appCodec, keys[ibcfeetypes.StoreKey],
-		app.IBCKeeper.ChannelKeeper, // may be replaced with IBC middleware
-		app.IBCKeeper.ChannelKeeper,
-		&app.IBCKeeper.PortKeeper, app.AccountKeeper, app.BankKeeper,
-	)
-
 	// Create Transfer Keepers
 	app.TransferKeeper = ibctransferkeeper.NewKeeper(
 		appCodec,
 		keys[ibctransfertypes.StoreKey],
 		app.GetSubspace(ibctransfertypes.ModuleName),
-		app.IBCFeeKeeper, // ISC4 Wrapper: fee IBC middleware
+		app.HooksICS4Wrapper,
 		app.IBCKeeper.ChannelKeeper,
 		&app.IBCKeeper.PortKeeper,
 		app.AccountKeeper,
 		app.BankKeeper,
 		scopedTransferKeeper,
 	)
+	transferIBCModule := transfer.NewIBCModule(app.TransferKeeper)
 
-	app.RouterKeeper.SetTransferKeeper(app.TransferKeeper)
+	// Hooks Middleware
+	hooksTransferStack := ibchooks.NewIBCMiddleware(&transferIBCModule, &app.HooksICS4Wrapper)
 
-	// ICA Host keeper
-	app.ICAHostKeeper = icahostkeeper.NewKeeper(
-		appCodec, keys[icahosttypes.StoreKey], app.GetSubspace(icahosttypes.SubModuleName),
-		app.IBCFeeKeeper, // use ics29 fee as ics4Wrapper in middleware stack
-		app.IBCKeeper.ChannelKeeper, &app.IBCKeeper.PortKeeper,
-		app.AccountKeeper, scopedICAHostKeeper, app.MsgServiceRouter(),
+	app.RouterKeeper = *routerkeeper.NewKeeper(
+		appCodec,
+		app.keys[routertypes.StoreKey],
+		app.GetSubspace(routertypes.ModuleName),
+		app.TransferKeeper,
+		app.IBCKeeper.ChannelKeeper,
+		app.DistrKeeper,
+		app.BankKeeper,
+		app.IBCKeeper.ChannelKeeper,
 	)
-	icaModule := ica.NewAppModule(nil, &app.ICAHostKeeper)
+	pmfTransferStack := router.NewIBCMiddleware(hooksTransferStack,
+		&app.RouterKeeper,
+		5,
+		routerkeeper.DefaultForwardTransferPacketTimeoutTimestamp,
+		routerkeeper.DefaultRefundTransferPacketTimeoutTimestamp,
+	)
+	app.TransferStack = &pmfTransferStack
```

Comment:
Your IBC transfer stack remove the ``IBCFee`` and place the ``hooks middeware`` before ``PFM`` . Can you explain why ? Because I think the ``hooks`` often have the final say

Reply:
Seems like the error from IBC was shadowed by a success response from a different module. The image below shows on the left the new message you will receive with the error and on the right is the old message.
![image](https://github.com/White-Whale-Defi-Platform/migaloo-chain/assets/49301655/4868e33c-369d-409b-b0d1-ccdb0bc225c0)
> Files with the messages attached below:
[right.txt](https://github.com/White-Whale-Defi-Platform/migaloo-chain/files/13247121/right.txt)
[left.txt](https://github.com/White-Whale-Defi-Platform/migaloo-chain/files/13247123/left.txt)


If you want to reproduce the issue try using the scripts from the folder scripts/tests

Reply:
I saw that the test scripts in increment.sh have the ``callback with timeout`` case. I think we should have a case where:
Chain A sends ibc hooks tx to Chain B, B response error to chain A.

Reply:
What if we still keep the ``IBCFee``? Is the problem come from the order of ``hooks middleware`` and ``PFM`` ? Or it include IBCFee

Reply:
If you want to reproduce the issue add the following `IBC_HOOK_RES=$($BINARY tx ibc-transfer transfer transfer channel-0 $CONTRACT_ADDRESS 1uwhale --memo='{"contract":"migaloo1v7cuk85rgfrsegsegsslre0lhgrvq28gqtlyms6","msg":{"execute_msgs":{"msgs":[{"msg":{"wasm":{"instantiate":{"code_id":2000, "msg": "eyJhbGxvd19jcm9zc19jaGFpbl9tc2dzIjogdHJ1ZX0=", "funds":[], "label":"Global proxy"}}}}]}}}}' --chain-id test-1 --home $CHAIN_DIR/test-1 --node tcp://localhost:16657 --keyring-backend test --from $WALLET_1 --broadcast-mode block -y -o json)` to the `increment.sh` I have tested it manually.

Reply:
Please feel free to fork this changes and apply any changes you think are important, you can experiment modifying the transfer stack if you want

Reply:
When I run the test locally, It still produce the same result. 
[ack.txt](https://github.com/White-Whale-Defi-Platform/migaloo-chain/files/13249613/ack.txt)

Reply:
@emidev98 can you show me how to pass the reproduced issue ?

Reply:
i figured the current stack on the main doesn't include ibc hooks middleware is the main issue, so we just need a little fix in ibc stack. But thanks for your work.

Reply:
I am happy to hear you found the issue, thanks for the info! 
If you don't need this PR anymore feel free to close it 🎉